docs/source/concepts/ledger-model/ledger-exceptions.rst

```diff
@@ -0,0 +1,288 @@
+   Node         ::= Action | Rollback
+   Rollback     ::= 'Rollback' Transaction
+   Action       ::= 'Create' contract
+                  | 'Exercise' party* contract Kind Transaction
+                  | 'Fetch' party* contract
+                  | 'NoSuchKey' key
+   Kind         ::= 'Consuming' | 'NonConsuming'
+
+Note that `Action` and `Kind` have the same definition as before, but
+since `Transaction` may now contain rollback nodes, this means that an
+`Exercise` action may have a rollback node as one of its consequences.
+For example, ...
+
+..
+   TODO: Add example of an exercise containing a rollback node
+   with diagram. 'Alice orders her bank to "withdraw 1000 USD,
+   or, if it fails, to withdraw 500 USD instead".'
+
+In addition, rollback nodes may be nested. This corresponds to a possible
+case where multiple exceptions are raised and caught within the same
+transaction.
+
+Integrity
++++++++++
+
+The notion of a ledger action coming "after" another ledger action must
+be revised in the presence of rollback nodes. It is not enough to traverse
+the transaction tree in prefix order, because the actions under a rollback
+were rolled back.
+
+For example, a contract may be consumed by an exercise under a rollback node,
+and immediately again after the rollback node. This is allowed because the
+exercise was rolled back, and this does not represent a "double spend" of
+the same contract.
+
+We define the "after" relation as a partial order, rather than a total order,
+on all the nodes (ledger actions and rollback nodes) of a transaction. This
+relation is generated by reflexivity, transitivity, and the following
+additional rules:
+
+#. If `node1` is a parent of `node2`, then `node2` comes after `node1`.
+
+#. If `node1` is a ledger action, and both `node1` and `node2` are
+   top-level nodes such that `node1` precedes `node2` in the list of
+   top-level nodes, then `node2` comes after `node1`.
+
+#. Similarly, if `node1` is a ledger action, and both `node1` and `node2`
+   share a parent (whether it be a rollback node or an exercise action)
+   such that `node1` precedes `node2` in the list of children, then
+   `node2` comes after `node1`.
+
+With this modified "after" relation, the notion of consistency remains the
+same. Note that under this definition, a rollback node can be "after" its
+siblings, but never "before". The only time a rollback node comes "before"
+another node, is when that other node is a child or descendant of the
+rollback node.
+
+So in the example of a consuming exercise in a rollback node, followed by a
+consuming exercise on the same contract outside of the rollback node, neither
+exercise comes "after" the other. They are part of separate continuities,
+one of which was rolled back, and so they don't break the ledger integrity.
+
+..
+   TODO: Add diagrams to demonstrate the "forking".
+
+Authorization
++++++++++++++
+
+Since they are not ledger actions, rollback nodes do not have authorizers
```

Comment:
You could also relate this to the existing definition in two ways:
1. a commit/ledger is well-authorized exactly when the same ledger with `Rollback` nodes removed (absorbed into their parent transactions) is well-authorized according to the old definition
2. define "top-level action" and "consequence" to "see through" `Rollback` nodes, and then just keep the existing rules.

Reply:
Added a link and made the connection clearer.